src/core/thread/mle.cpp

```diff
@@ -3350,16 +3350,20 @@ void Mle::HandleChildIdResponse(RxInfo &aRxInfo)
 
 void Mle::HandleChildUpdateRequest(RxInfo &aRxInfo)
 {
+    VerifyOrExit(IsAttached());
```

Comment:
Maybe there's a different way I should handle it, but the reset changes right now utilize the child update request to know when to try to reconnect to a previous parent. i.e. while waiting to restore, a previously child role device will listen for child update requests from it's old parent to know when to try to rejoin. This design requires handling the update request in the detached state.

It could be that we just add different handling in that case though to make it explicit.

Reply:
Yes, I think this is still possible. We can check if a "Child Update Request" was received from the former parent while the child is in restoring mode. We can add this before `VerifyOrExit(IsAttached());`.

The desired behavior we are adding in this PR is that a detached device should not respond to a "Child Update Request," regardless of its state (e.g., restoring or attaching).